fuzz/fuzz_targets/cranelift-compile.rs

```diff
@@ -0,0 +1,131 @@
+#![no_main]
+
+use libfuzzer_sys::{arbitrary, fuzz_target};
+use once_cell::sync::Lazy;
+use std::fmt;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
+use target_lexicon::Architecture;
+
+use cranelift_codegen::ir;
+use cranelift_codegen::isa;
+use cranelift_codegen::settings;
+use cranelift_filetests::function_runner::TestFileCompiler;
+use cranelift_fuzzgen::*;
+
+/// Gather statistics about the fuzzer executions
+struct Statistics {
+    /// Total amount of runs that we tried to compile.
+    pub total_runs: AtomicU64,
+
+    /// How many runs were successful?
+    /// This is also how many runs were run in the backend
+    pub run_result_success: AtomicU64,
+}
+
+impl Statistics {
+    pub fn print(&self, total_runs: u64) {
+        // We get valid_inputs as a param since we already loaded it previously.
+        let run_result_success = self.run_result_success.load(Ordering::SeqCst);
+
+        println!("== FuzzGen Statistics  ====================");
+        println!("Total Runs: {}", total_runs);
+        println!(
+            "Successful Runs: {} ({:.1}% of Total Runs)",
+            run_result_success,
+            (run_result_success as f64 / total_runs as f64) * 100.0
+        );
+    }
+}
+
+impl Default for Statistics {
+    fn default() -> Self {
+        Self {
+            total_runs: AtomicU64::new(0),
+            run_result_success: AtomicU64::new(0),
+        }
+    }
+}
+
+static STATISTICS: Lazy<Statistics> = Lazy::new(Statistics::default);
+
+struct CompileTest {
+    isa: Box<dyn isa::TargetIsa>,
+    func: ir::Function,
+}
```

Comment:
I think we might be able to reuse this for the icache fuzzer as well. Feel free to ignore it if it's too much work to change that.

Reply:
That's a great point! I'd be happy to pull this type back into cranelift-fuzzgen so that we can use it in both targets :+1:

Reply:
I've added a new struct `FunctionWithIsa` in cranelift-fuzzgen, and will rework cranelift-icache in a separate PR once this one has landed.